### ISSUE TITLE
Segfault when z not specified on command line

### ISSUE THREAD

**Issue by alexander-mead** [miknab/EuclidEmulator2#6]
If I run the emulator without specifying an output z, for example:
```
./ee2.exe -b 0.05 -m 0.3 -s 0. -n 0.96 -H 0.7 -W -1 -w 0. -A 2.1e-9
```
I get an unhelpful segmentation fault:
```
The current cosmology is mapped to:

        tf(Omega_b) = 4.4408920985e-16
        tf(Omega_m) = -0.25
        tf(Sum m_nu) = -1
        tf(n_s) = -1.44328993201e-15
        tf(h) = 0.5
        tf(w_0) = 0
        tf(w_a) = 0
        tf(A_s) = 4.4408920985e-16

Segmentation fault
```
Things work perfectly if instead I do:
```
./ee2.exe -b 0.05 -m 0.3 -s 0. -n 0.96 -H 0.7 -W -1 -w 0. -A 2.1e-9 -z 0.
```

**Comment by miknab** [comment]
Hi Alexander,

Yes, that's correct. There is no default value set for z which is intentional. It would probably be a good idea, though, to print a meaningful error message in this case. Thanks for the hint.

Is this what you were going for?

Regards,
Mischa

**Comment by alexander-mead** [comment]
Yeah, absolutely, sorry for not making that clearer. It just took me a couple of minutes to figure out what I had done wrong and I thought a nice error message (like those that appear when the cosmology is not fully specified) would have saved me some time.

**Comment by miknab** [comment]
Yes, I absolutely agree. I will add this to the next update. Thanks again for the hint.

**Comment by alexander-mead** [comment]
One further thing. I noticed that I also get a segmentation fault if I run the code from a directory other than the `EuclidEmulator2/` directory (where my install is). I guess this is because the code can't find some files that it needs. It might be nice to have a graceful exit in this case too.

**Comment by miknab** [comment]
Hm... ok, do you also have just an error message in mind here or something else? It's good to see that people are stress-testing my code :-D. Thanks for the feedback. I really appreciate it.

**Comment by alexander-mead** [comment]
Yeah, exactly. Just something like `error, files not found` or something more helpful than `Segmentation fault`. It just took me a bit of hunting to figure out what was going wrong.

I'll keep stress testing, but at the moment everything seems to be working well; it was very easy for me to install and get running.

**Comment by miknab** [comment]
Great... I love to hear that. Would you care sharing the specifics of your system/environment such that I can add it to the list of tested environments?

**Comment by alexander-mead** [comment]
Sure, I'm on Mac OS 10.15.7 (Catalina) with gsl (2.5) and g++ (9.2.0_3) installed via Homebrew.

**Comment by miknab** [comment]
Hi Alexander,

I have just added the error message for missing redshift values as you suggested. So far it's only available in the version sitting in the development branch, though. I'll pull it into the master branch soon.

PS: Also, very recently a pip installable python wrapper was added (courtesy of Pedro Carrilho). If you feel like it please feel free to update your installation of EE2 and play around with it. We are currently gathering feedback and would highly appreciate yours.
